core/version_test.go

```diff
@@ -42,3 +43,66 @@ func TestCannotParseBlockVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportedBlockVersion(t *testing.T) {
+	testVersions := []struct {
+		block  semver.Version
+		latest semver.Version
+	}{
+		{
+			block:  *core.LatestVer,
+			latest: *core.LatestVer,
+		},
+		{
+			block:  core.LatestVer.IncPatch(),
+			latest: *core.LatestVer,
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncPatch(),
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncMinor(),
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncMajor(),
+		},
```

Comment:
Add comments for `TestSupportedBlockVersion` test cases

```suggestion
		{
			// Block and latest are the same version (e.g., 0.14.0 == 0.14.0)
			block:  *core.LatestVer,
			latest: *core.LatestVer,
		},
		{
			// Block is newer patch than latest (e.g., 0.14.1 > 0.14.0)
			block:  core.LatestVer.IncPatch(),
			latest: *core.LatestVer,
		},
		{
			// Block is older patch than latest (e.g., 0.14.0 < 0.14.1)
			block:  *core.LatestVer,
			latest: core.LatestVer.IncPatch(),
		},
		{
			// Block is older minor than latest (e.g., 0.14.0 < 0.15.0)
			block:  *core.LatestVer,
			latest: core.LatestVer.IncMinor(),
		},
		{
			// Block is older major than latest (e.g., 0.14.0 < 1.0.0)
			block:  *core.LatestVer,
			latest: core.LatestVer.IncMajor(),
		},
```